### PR TITLE
Enable -Xjvm-default=all for the Kotlin DSL

### PR DESCRIPTION
**Pull request by eskatos**
See https://blog.jetbrains.com/kotlin/2020/07/kotlin-1-4-m3-generating-default-methods-in-interfaces/

Issue https://github.com/gradle/gradle/issues/14875

### CODE REVIEW

**Comment by eskatos** [comment]
@bot-gradle test RFN

**Review by bamboo — APPROVED** [review]

**Review by bamboo — CHANGES_REQUESTED** [review]
We should include at least one integration test. Other than that, it looks good to me!

**Comment by eskatos** [comment]
@bot-gradle test this

File: subprojects/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/KotlinDslJvmDefaultIntegrationTest.kt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+import org.gradle.integtests.fixtures.RepoScriptBlockUtil.mavenCentralRepository
+import org.gradle.test.fixtures.dsl.GradleDsl.KOTLIN
+import org.junit.Test
+
+
+class KotlinDslJvmDefaultIntegrationTest : AbstractPluginIntegrationTest() {
+
+    @Test
+    fun `kotlin-dsl java and groovy consumers can use kotlin interface default methods directly`() {
```
Comment (bamboo) [inline subprojects/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/KotlinDslJvmDefaultIntegrationTest.kt:27]: ❌ Nice! We should also have a test for a Kotlin DSL script consumer implementing a Java interface with default methods.

**Review by bamboo — CHANGES_REQUESTED** [review]

**Comment by eskatos** [comment]
@bot-gradle test this

**Review by bamboo — APPROVED** [review]

**Comment by eskatos** [comment]
@bot-gradle test this

**Review by bamboo — APPROVED** [review]

**Comment by eskatos** [comment]
@bot-gradle test this

**Pull request merged**